packages/sst/src/constructs/NextjsSite.ts

```diff
@@ -282,6 +282,8 @@ export class NextjsSite extends SsrSite {
       headerBehavior: cloudfront.CacheHeaderBehavior.allowList(
         // required by image optimization request
         "accept",
+        // required by middleware for <Link /> routing in place
+        "rsc",
```

Comment:
Isn't there more headers, like router state that might be needed as well? https://github.com/vercel/next.js/blob/canary/packages/next/src/client/components/app-router-headers.ts

Reply:
Yep there's a few more headers but I didn't see any difference with or without them so I only included what was necessary.

I may have missed something though and it probably wouldn't hurt to include the others. But one thing to keep on mind is that there's a max number of headers you can set.

Reply:
It might have something to do with prefetching links on hover. I'll test it out in a few hours.

Reply:
Looking at the code, _I think_ the router state is used to know how much of the RSC layout components that need to be sent to the client.

Reply:
So by not allowing it will lead to entire layout tree must be sent on every navigation.

Reply:
Thanks for looking into it! Updated.

Reply:
Hmm.. behavior wise, I don't think anything changes though. My `<Link />`s are getting prefetched immediately, while local development only loads them when I hover over the links.

Reply:
Yeah it's all pretty subtle. Would be nice to have docs for this from nextjs side.

Reply:
Good idea, I'll add links to the nextjs source in a bit